tests/import_test.go

```diff
@@ -1770,30 +1770,29 @@ var _ = Describe("Import populator", func() {
 		verifyCleanup(pvc)
 
 		By("Making PV available")
-		Eventually(func() bool {
+		Eventually(func() v1.PersistentVolumePhase {
 			pv, err := f.K8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			Expect(pv.Spec.ClaimRef.Namespace).To(Equal(pvc.Namespace))
 			Expect(pv.Spec.ClaimRef.Name).To(Equal(pvc.Name))
-			if pv.Status.Phase == v1.VolumeAvailable {
-				return true
+			if pv.Status.Phase != v1.VolumeAvailable {
+				pv.Spec.ClaimRef.ResourceVersion = ""
+				pv.Spec.ClaimRef.UID = ""
+				_, err = f.K8sClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
Expecting inside an eventually block may fail the test earlier than we want to
to get around this you can do this:
https://github.com/kubevirt/containerized-data-importer/pull/2696/files#diff-e3e6528c8b5d4860f4fd89d38f0e0babb9cc0e9147c0b0b28be532e5e692c182R99

Reply:
So I didn't change that particular thing, I just cleaned up the return value a bit so it is easier to see what actually failed. This expect if it return false it is because the update failed. Which should fail the test right there.